### PR TITLE
feat(azure): support different merge strategies for autocomplete

### LINKED ISSUE #4514: More options needed when setting autocomplete on azure devops.
**What would you like Renovate to be able to do?**
Control the options used when a dev ops pull request auto completes, currently this is hard coded to squash merge and we don't use squash merges in our setup. (this is a policy decision that we are not changing)

**Describe the solution you'd like**
I would like an option that lets me control the options passed in this azure api call.
https://github.com/renovatebot/renovate/blob/cc1233b7d6f344df983b0fdf155dfa603417fe39/lib/platform/azure/index.ts#L380
specifically the merge type.
see :  here for the options.
https://github.com/microsoft/azure-devops-node-api/blob/f21b5bf4c9b49d53c81eedf771ccd4165b58d1d0/api/interfaces/GitInterfaces.ts#L1732

**Describe alternatives you've considered**
Not using auto complete 

**Additional context**

### PR DESCRIPTION
**Pull request by JamieMagee**
Fixes #4514

Also removed some JSDoc that was left from pre-TypeScript migration

@stocksr Could you take a look and see if this works for you?

### CODE REVIEW

**Comment by stocksr** [comment]
Happy to confirm that this change works for me.

in case it is important: I tested this using docker

**Comment by JamieMagee** [comment]
Super! I just need to get the test coverage back to 100% and we can merge this.

File: lib/platform/azure/azure-helper.ts [hunk 1]
```diff
@@ -119,12 +98,6 @@ export async function getChanges(
   return changes;
 }
 
-/**
- * if no branchName, look globaly
```
Comment (ViceIce) [inline lib/platform/azure/azure-helper.ts]: This line can be left here as docu. We should add more of those 😅

File: lib/platform/azure/azure-helper.ts [hunk 2]
```diff
@@ -271,3 +236,24 @@ export function getProjectAndRepo(str: string) {
...
+  repoId: string,
+  project: string
+): Promise<GitPullRequestMergeStrategy> {
+  const policyConfigurations = (await (await azureApi.policyApi()).getPolicyConfigurations(
+    project
+  ))
+    .filter(
+      p =>
+        p.settings.scope.some(s => s.repositoryId === repoId) &&
+        p.type.id === 'fa4e907d-c16b-4a4c-9dfa-4916e5d171ab' // Magic GUID for merge strategy policy configurations
```
[the remark this answers is not in the document]
Comment (JamieMagee) [reply lib/platform/azure/azure-helper.ts]: I can't see us adding any more in the future, but I agree that extracting to a constant should clarify what this GUID is.

[the remark this answers is not in the document]
Comment (rarkins) [reply lib/platform/azure/azure-helper.ts]: Awaiting this

**Review by viceice — APPROVED** [review]

**Pull request merged**